### PR TITLE
M4 IMPL - PR #2 - Kafka event POJOs, topic constants, and common envelope in common

### PR DESCRIPTION
**Pull request by Satvik2810**
# Pull Request

## Summary
Establishes all shared Kafka contracts in `common` (Phase 0 of M4 implementation) and extends the M4 design with two new fulfillment paths (self-drop pickup, hub-collect delivery), full handover verification design (OTP at pickup, QR scan at every internal custody transfer), and label management documentation.

---

## What Changed
- **`common/` — Kafka contracts:** `ShipmentState` (27 values), domain enums (`CustomerType`, `DeliveryType`, `PaymentMode`, `PickupType`, `DropType`), 9 Kafka topic constants, 7 per-topic event type enums (`DaEventType`, `HubEventType`, `ScanEventType`, `FlightEventType`, `CronEventType`, `ExceptionsEventType`, `ShipmentEventType`), and 4 event POJOs (`BaseShipmentEvent`, `ShipmentCreatedEvent`, `ShipmentStateChangedEvent`, `ShipmentCancelledEvent`) — all forward-compatible with `@JsonIgnoreProperties(ignoreUnknown = true)`
- **`M4-STATE-MACHINE.md`:** Complete rewrite of the visual flow diagram — now shows all path branches (DA_PICKUP, SELF_DROP, DA_DELIVERY, HUB_COLLECT), OTP and QR annotations at every transition, and a verification legend; states reference table expanded to all 27 states with a "Verification to enter" column
- **`M4-ORDERS-DESIGN.md`:** Added §6.6 (label lifecycle — when generated by M8, when physically attached by DA or customer), §6.7 (handover verification table + OTP lifecycle), two new OTP internal endpoints in §7.5, corrected §9.2 to remove `PICKUP_COMPLETED` from M4's consumed events, added OD-8 and OD-9 to open decisions; state count corrected 24 → 27 throughout
- **`M4-IMPL-PLAN.md`:** PR #9 expanded with `PickupOtpService`, `pickup_otps` migration, and OTP endpoints; state/test matrix counts corrected to 27
- **`docs/MODULES.md`:** `da.cron_handoff_completed` → `da.van_handoff_completed`
- **Key architectural decision:** `PICKED_UP` is now triggered by the OTP verify HTTP endpoint, not by `PICKUP_COMPLETED` Kafka event — documented in §6.7 and §9.2

---

## Why This Change?
Phase 0 contracts must exist in `common` before any module can write a Kafka producer or consumer. This PR publishes the shared event schemas and topic constants that unblock M2, M3, M5, M7, M8, M9, M10, and M11 teams. The design additions (self-drop, hub-collect, OTP/QR verification) were decisions made during design review that needed to be captured before implementation begins.

---

## Testing
- [x] Tested locally
- [x] Edge cases considered
- [x] No breaking changes introduced

### Test Evidence

$ mvn clean install -pl common

[INFO] Scanning for projects...
[INFO]
[INFO] -------------------------< com.oneday:common >--------------------------
[INFO] Building M0 - Common Shared Infrastructure 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- clean:3.3.2:clean (default-clean) @ common ---
[INFO] Deleting .../common/target
[INFO]
[INFO] --- compiler:3.11.0:compile (default-compile) @ common ---
[INFO] Changes detected - recompiling the module! :source
[INFO] Compiling 20 source files with javac [debug release 21] to target/classes
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::objectFieldOffset has been called by lombok.permit.Permit
WARNING: sun.misc.Unsafe::objectFieldOffset will be removed in a future release
[INFO]
[INFO] --- surefire:3.1.2:test (default-test) @ common ---
[INFO] No tests to run.
[INFO]
[INFO] --- jar:3.3.0:jar (default-jar) @ common ---
[INFO] Building jar: .../common/target/common-1.0.0-SNAPSHOT.jar
[INFO]
[INFO] --- install:3.1.1:install (default-install) @ common ---
[INFO] Installing common-1.0.0-SNAPSHOT.jar to local Maven repository
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.825 s
[INFO] Finished at: 2026-05-17T21:45:44+05:30
[INFO] ------------------------------------------------------------------------

> The Lombok `sun.misc.Unsafe` warning is a known JDK 21 deprecation from Lombok itself — present across the entire codebase and not introduced by this PR.

---

## Impact
Unblocks all other modules — M2, M3, M5, M7, M8, M9, M10, M11 can now implement against `common` Kafka contracts without waiting for M4's business logic.

Two open decisions require team input before implementation proceeds:
- **OD-8** — delivery verification mechanism (OTP mirrors pickup, vs QR scan) — blocks M5 DA app
- **OD-9** — event for `DEST_HUB_PROCESSING → AWAITING_HUB_COLLECT` — blocks M7 hub implementation

### Areas Affected
- [x] Backend
- [ ] Frontend
- [ ] Routing
- [ ] Infra
- [ ] Database

### Modules Affected
- [ ] M1 — Auth
- [ ] M2 — Pricing
- [ ] M3 — Grid
- [x] M4 — Orders
- [ ] M5 — Dispatch
- [ ] M6 — Routing
- [ ] M7 — Hub
- [ ] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [x] NA — `common` (shared contracts)

---

## Risks / Concerns
- `PICKUP_COMPLETED` (`DaEventType`) is still produced by M5 for M10's consumption but M4 no longer consumes it for state transitions — M5 team must be aware before implementing their Kafka producer
- `HubEventType` intentionally has no event for `DEST_HUB_PROCESSING → AWAITING_HUB_COLLECT` — known gap pending OD-9 resolution; M7 team should not implement against this transition until OD-9 is decided

---

## Checklist
- [x] Code follows project conventions (package layout, no cross-module internal imports)
- [x] Self-review completed
- [x] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [x] No sensitive data or credentials exposed
- [x] Documentation updated if behaviour changed
- [x] Ready for review

### CODE REVIEW

File: common/src/main/java/com/oneday/common/kafka/enums/ScanEventType.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.oneday.common.kafka.enums;
+
+public enum ScanEventType {
+    HUB_ORIGIN_IN,          // DA pickup path: HANDED_TO_PICKUP_VAN → AT_ORIGIN_HUB
+    SELF_DROP_ACCEPTED,     // self-drop path: AWAITING_SELF_DROP → AT_ORIGIN_HUB
+    GHA_ACCEPTANCE,
+    HUB_DEST_IN,
+    LABEL_GENERATED,
+    HUB_COLLECT_COMPLETED   // hub-collect path: AWAITING_HUB_COLLECT → HUB_COLLECTED
```
Comment (Sidarthapati) [inline common/src/main/java/com/oneday/common/kafka/enums/ScanEventType.java:9]: Are these the only scan events possible? What about first mile, handoff to van, i know we can change this down the line so not a blocker

**Review by Sidarthapati — APPROVED** [review]

File: common/src/main/java/com/oneday/common/kafka/enums/ScanEventType.java (hunk 1 above)
Comment (Satvik2810) [reply common/src/main/java/com/oneday/common/kafka/enums/ScanEventType.java:9]: ScanEventType = scans that M8 physically owns — the hub scan stations and GHA terminal. 

The other QR scans in the design doc (e.g. VAN_HANDOFF_COMPLETED, DROP_COLLECTED, DEPARTED_HUB, STAND_ASSIGNED) are owned by M5, M6, M7 respectively and live in their own enums.

Comment (Satvik2810) [reply common/src/main/java/com/oneday/common/kafka/enums/ScanEventType.java:9]: Screenshot 2026-05-19 at 10 14 43 PM

Comment (Satvik2810) [reply common/src/main/java/com/oneday/common/kafka/enums/ScanEventType.java:9]: This is present in the design doc
